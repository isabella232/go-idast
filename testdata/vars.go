package vars

var A = 1
var b = A + 2
var c = "foo"
