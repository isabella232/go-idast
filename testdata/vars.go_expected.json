 File            | package vars\n\nvar A = 1\nvar  | 
 Ident           | vars                            | Name/Ident
 GenDecl         | var A = 1                       | Decls/0/GenDecl
 ValueSpec       | A = 1                           | Decls/0/GenDecl/Specs/0/ValueSpec
 Ident           | A                               | Decls/0/GenDecl/Specs/0/ValueSpec/Names/A/Ident
 BasicLit        | 1                               | Decls/0/GenDecl/Specs/0/ValueSpec/Values/0/BasicLit
 GenDecl         | var b = A + 2                   | Decls/1/GenDecl
 ValueSpec       | b = A + 2                       | Decls/1/GenDecl/Specs/0/ValueSpec
 Ident           | b                               | Decls/1/GenDecl/Specs/0/ValueSpec/Names/b/Ident
 BinaryExpr      | A + 2                           | Decls/1/GenDecl/Specs/0/ValueSpec/Values/0/BinaryExpr
 Ident           | A                               | Decls/1/GenDecl/Specs/0/ValueSpec/Values/0/BinaryExpr/X/Ident
 BasicLit        | 2                               | Decls/1/GenDecl/Specs/0/ValueSpec/Values/0/BinaryExpr/Y/BasicLit
 GenDecl         | var c = "foo"                   | Decls/2/GenDecl
 ValueSpec       | c = "foo"                       | Decls/2/GenDecl/Specs/0/ValueSpec
 Ident           | c                               | Decls/2/GenDecl/Specs/0/ValueSpec/Names/c/Ident
 BasicLit        | "foo"                           | Decls/2/GenDecl/Specs/0/ValueSpec/Values/0/BasicLit
